src/client/components/hud/Initiative.tsx

```diff
@@ -252,17 +259,88 @@ export function InitiativeHUD() {
 
   const focusTokenOnTurnStart = useRRSettings()[0].focusTokenOnTurnStart;
 
+  const updateConcentration = useCallback(
+    (
+      character: RRCharacter,
+      updater: React.SetStateAction<RRCharacter["currentlyConcentratingOn"]>
+    ) => {
+      dispatch((state) => {
+        const old =
+          state.characters.entities[character.id]?.currentlyConcentratingOn;
+
+        if (old === undefined) return [];
+        const newConcentration =
+          typeof updater === "function" ? updater(old) : updater;
+
+        return {
+          actions: [
+            characterUpdate({
+              id: character.id,
+              changes: { currentlyConcentratingOn: newConcentration },
+            }),
+          ],
+          optimisticKey: "concentration",
+          syncToServerThrottle: 0,
+        };
+      });
+    },
+    [dispatch]
+  );
+
+  const alertPlayer = useCallback(
+    async (characterName: string, concentrationObject: string) => {
+      await alert(
+        `${characterName}'s concentration on ${concentrationObject} has expired.`
+      );
+    },
+    [alert]
+  );
+
   useEffect(() => {
     if (
-      focusTokenOnTurnStart &&
       currentRow &&
       currentRow.id !== lastRowId.current &&
       currentRow.type === "character"
     ) {
-      setSelection(currentRow.characterIds);
+      if (focusTokenOnTurnStart) setSelection(currentRow.characterIds);
+      currentRow.characterIds.forEach((characterId) => {
+        const character = charactersRef.current.entities[characterId];
+        if (character?.currentlyConcentratingOn) {
+          updateConcentration(character, {
+            ...character.currentlyConcentratingOn,
+            roundsLeft: character.currentlyConcentratingOn.roundsLeft - 1,
```

Comment:
I think this won't work as you expect. Since this code runs on the client, it will be executed on each player's computer for each character. Thus the concentrations rounds will be decremented by the number of players connected :D #DNDHardMode
I think we need to move this automatic updating of the concentration to the server instead.

Reply:
You're right, there is a race condition, which makes it work again. When you pass a function to dispatch, the function runs on the clients and the result is sent to the server. Thus, on every client, we take the current state, increment it by one, and then send the resulting state as an update to the server. While this causes it to 'work' correctly in this case, it would lead to issues in other cases. (imagine a shared counter between the players with a button to increment it. Passing a function to dispatch like currently done in this PR would lead to race conditions if multiple players attempt to update it at the same time)
This is one of the reasons why we should generally try to avoid passing a function to dispatch, I think, and instead define custom redux actions like 'decrementSpellRounds' for cases like  this. Defining the action moves the Maths from the clients to the implementation of the action on the server, and this avoid the race condition. 

Having said all that, the case of this PR is special because we want the updating to happen automatically based on other state updates. The database world equivalent would be to define a TRIGGER, I think. So to do this, all we need to do is to subscribe to state updates on the server (store.subscribe()), keep track of when the initiative entry changes, similar to how we do it on the client, and then store.dispatch() an action/actions to decrement the durations.